receiver/azuremonitorreceiver/scraper_batch.go

```diff
@@ -307,6 +307,45 @@ func (s *azureBatchScraper) getResourcesAndTypes(ctx context.Context, subscripti
 	maps.Copy(s.resourceTypes[subscriptionID], resourceTypes)
 }
 
+// hackResources is a workaround specially done for the storageAccount metrics.
+// Every StorageAccount resources have some implicit sub resources (/blobServices/default, fileServices/default, etc...) that are not returned by the API.
+// We need to add them manually to the resources and resourceTypes map.
+// Note that we do that hack only if the user has asked the sub resource types explicitly in the services config.
+// Example:
+// For each resource with id .../Microsoft.Storage/storageAccount/myResource of type Microsoft.Storage/storageAccount,
+// It will create a fake resource with id .../Microsoft.Storage/storageAccount/myResource/blobServices/default of type Microsoft.Storage/storageAccounts/blobServices.
+// TODO: duplicate
```

Comment:
Note that I have a TODO duplicate on several methods of the two scraper. I would like to address it in another PR to not pollute. I promise I will do it right after this PR will be merged.